src/wallet/api.rs

```diff
@@ -1267,36 +1270,71 @@ impl Wallet {
-            return Ok(Vec::new());
-        }
+        // Get regular and swap UTXOs separately
+        let available_regular_utxos: Vec<(ListUnspentResultEntry, UTXOSpendInfo)> =
+            filter_locked(self.list_descriptor_utxo_spend_info()?);
+        let available_swap_utxos: Vec<(ListUnspentResultEntry, UTXOSpendInfo)> =
+            filter_locked(self.list_swept_incoming_swap_utxos()?);
 
-        // Assert: fidelity, outgoing swapcoins, hashlock and timelock coins are not included
+        // Assert that no non-spendable UTXOs are included after filtering
         assert!(
-            unspents.iter().all(|(_, spend_info)| !matches!(
-                spend_info,
-                UTXOSpendInfo::FidelityBondCoin { .. }
-                    | UTXOSpendInfo::OutgoingSwapCoin { .. }
-                    | UTXOSpendInfo::TimelockContract { .. }
-                    | UTXOSpendInfo::HashlockContract { .. }
-            )),
-            "Fidelity, Outgoing Swapcoins, Hashlock and Timelock coins are not included in coin selection"
-        );
+       available_regular_utxos.iter().chain(available_swap_utxos.iter()).all(|(_, spend_info)| !matches!(
+           spend_info,
+           UTXOSpendInfo::FidelityBondCoin { .. }
+               | UTXOSpendInfo::OutgoingSwapCoin { .. }
+               | UTXOSpendInfo::TimelockContract { .. }
+               | UTXOSpendInfo::HashlockContract { .. }
+       )),
+       "Fidelity, Outgoing Swapcoins, Hashlock and Timelock coins are not included in coin selection"
+   );
+
+        if available_regular_utxos.is_empty() && available_swap_utxos.is_empty() {
+            log::error!("No spendable UTXOs available");
+            return Err(WalletError::InsufficientFund {
+                available: 0,
+                required: amount.to_sat(),
+            });
+        }
+
+        // Calculate totals for each type
+        let regular_total: u64 = available_regular_utxos
+            .iter()
+            .map(|(utxo, _)| utxo.amount.to_sat())
+            .sum();
+        let swap_total: u64 = available_swap_utxos
+            .iter()
+            .map(|(utxo, _)| utxo.amount.to_sat())
+            .sum();
+        let target_sats = amount.to_sat();
+
+        // Determine which UTXO types can satisfy the target
+        let can_use_regular = target_sats <= regular_total;
+        let can_use_swap = target_sats <= swap_total;
+
+        // Early exit: Only proceed if at least one type can satisfy the target
+        if !can_use_regular && !can_use_swap {
+            if regular_total + swap_total < target_sats {
+                return Err(WalletError::InsufficientFund {
+                    available: regular_total + swap_total,
+                    required: target_sats,
+                });
+            } else {
+                return Err(WalletError::Selection(
+                    rust_coinselect::types::SelectionError::NoSolutionFound,
+                ));
+            }
+        }
```

Comment:
only return insufficient funds on regular balance. As swap balance cannot be replinished easily.